src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/NrbfDecoder.cs

```diff
@@ -23,26 +24,25 @@ public static class NrbfDecoder
     /// </summary>
     /// <param name="bytes">The buffer to inspect.</param>
     /// <returns><see langword="true" /> if it starts with NRBF payload header; otherwise, <see langword="false" />.</returns>
-    public static bool StartsWithPayloadHeader(byte[] bytes)
+    public static bool StartsWithPayloadHeader(ReadOnlySpan<byte> bytes)
     {
-#if NET
-        ArgumentNullException.ThrowIfNull(bytes);
-#else
-        if (bytes is null)
+        if (bytes.Length < SerializedStreamHeaderRecord.Size || bytes[0] != (byte)SerializationRecordType.SerializedStreamHeader)
         {
-            throw new ArgumentNullException(nameof(bytes));
+            return false;
         }
-#endif
 
-        return bytes.Length >= SerializedStreamHeaderRecord.Size
-            && bytes[0] == (byte)SerializationRecordType.SerializedStreamHeader
-#if NET
-            && BinaryPrimitives.ReadInt32LittleEndian(bytes.AsSpan(9)) == SerializedStreamHeaderRecord.MajorVersion
-            && BinaryPrimitives.ReadInt32LittleEndian(bytes.AsSpan(13)) == SerializedStreamHeaderRecord.MinorVersion;
-#else
-            && BitConverter.ToInt32(bytes, 9) == SerializedStreamHeaderRecord.MajorVersion
-            && BitConverter.ToInt32(bytes, 13) == SerializedStreamHeaderRecord.MinorVersion;
-#endif
+        ReadOnlySpan<int> integers = MemoryMarshal.Cast<byte, int>(bytes.Slice(sizeof(byte), sizeof(int) * 4));
```

Comment:
This is not 100% portable code. `ReadOnlySpan<int>` that this creates is guaranteed to be misaligned and this package supports netstandard2.0. From https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.memorymarshal.cast: "This method is supported only on platforms that support misaligned memory access or when the memory block is aligned by other means."

I would keep the code as-is and just replaced `BitConverter.ToInt32` else part of `#if NET` with a local polyfill:

```
static int ReadInt32LittleEndian(ReadOnlySpan<byte> source)
   => source[0] + (source[1] << 8) + (source[2] << 16) + (source[3] << 24);
```